### DISCUSSION TITLE
Any way to listen to all Message and Voice Events?

### DISCUSSION THREAD

**Discussion by vinicius507** [andersfylling/disgord#437] [Q&A] [answered]
Greetings,

I'd like to know if there is any way to listen to ALL Messages and Voice Events without having to register the same handler multiple times.

Thanks in advance,
Vini.

**Comment by andersfylling** [comment]
when you say message, do u mean message create, message update and message delete events?

What does your current code look like?

**Reply by vinicius507** [reply]
Exactly. With Voice I just saw that `VoiceStateUpdate` registers a handler for all VoiceEvents (don't know how to filter per Guild, though).

Using some abstraction, you can think of it as something like this:

```go
func main() {
	var client = disgord.New(disgord.Config{
		BotToken: os.Getenv("BOT_TOKEN"),
		Logger:   log,
	})
	defer client.Gateway().StayConnectedUntilInterrupted()

	client.Gateway().
		WithContext(context.Background()).
		MessageCreate(func(s disgord.Session, data *disgord.MessageCreate) {
			events.TextMessage(session, data)
		})
	client.Gateway().
		WithContext(context.Background()).
		MessageUpdate(func(s disgord.Session, data *disgord.MessageUpdate) {
			events.TextMessage(session, data)
		})
	client.Gateway().
		WithContext(context.Background()).
		MessageDelete(func(s disgord.Session, data *disgord.MessageDelete) {
			events.TextMessage(session, data)
		})
}
```

EDIT: I don't need the message content, since the ability of receiving general text chat content will be removed with the implementation of Slash Commands in April. I just care about the events.

**Reply by andersfylling** [reply]
Oh, interesting use case of context. That's a design mistake on my part. Context has no effect on handlers. You have to use a Ctrl for that.

But yes those type of events are fundementally different, so there is no way to get them all in one handler.

**Comment by andersfylling** [comment, marked as answer]
In short: No. Each handler is uniquely mapped to a single discord event.

**Reply by vinicius507** [reply]
Thanks
